Sources/Diagnostics/DiagnosticsFileHandler.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  DiagnosticsFileHandler.swift
+//
+//  Created by Nacho Soto on 6/16/23.
+
+import Foundation
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+protocol DiagnosticsFileHandlerType: AnyObject {
+
+    func getEntries() async -> [DiagnosticsEvent]
+
+    func appendEvent(diagnosticsEvent: DiagnosticsEvent) async
+
+    func cleanSentDiagnostics(diagnosticsSentCount: Int) async
+
+    func emptyDiagnosticsFile() async
+
+}
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+actor DiagnosticsFileHandler: DiagnosticsFileHandlerType {
+
+    private let fileHandler: FileHandler
+
+    init?() {
+        do {
+            self.fileHandler = try .init(Self.diagnosticsFile)
+        } catch {
+            Logger.error("Initialization error: \(error.localizedDescription)")
+            return nil
+        }
+    }
+
+    init(_ fileHandler: FileHandler) {
+        self.fileHandler = fileHandler
+    }
+
+    func appendEvent(diagnosticsEvent: DiagnosticsEvent) async {
+        guard let jsonString = try? diagnosticsEvent.encodedJSON else {
+            Logger.error("Failed to serialize diagnostics event to JSON")
+            return
+        }
+
+        await fileHandler.append(line: jsonString)
+    }
+
+    func getEntries() async -> [DiagnosticsEvent] {
```

Comment:
I think this should be fine, but note that it shouldn't be necessarily needed to parse the entries back to a list of `DiagnosticsEntry`. We just need a list of jsons to send in the backend request. But I don't think this should be a performance issue, so we can do this 👍

Reply:
I see! Just noticed Android is also returning a stream of jsons